### ISSUE TITLE
Unreliable CMake module references

### ISSUE THREAD

**Issue by mpbelhorn** [ramkikannan/nmflibrary#1]
Hi Ramki,

The lines at

https://github.com/ramkikannan/planc/blob/a254454cdd738fa2669c6a427f3b9c0e0e7c0d66/common/cmake_aux/NMFLIB_FindARMA.cmake#L48-L50

are not generally available in portable builds. The armadillo CMake modules would exist for a local in-place source build of armadillo but are not generally available for builds of PLANC when armadillo is already installed.

I would recommend instead of directly including these modules (`ARMA_FindOpenBlas`, etc) to rely on CMake's builtin package finding mechanisms, something like ():

```
set(BLA_VENDOR OpenBLAS)  # force following two lines to look for OpenBlas only
find_package(BLAS REQUIRED)  # Set other constraints as necessary
find_package(LAPACK REQUIRED)
```

The stock FindBLAS.cmake and FindLAPACK.cmake modules should be quite sufficient (the `ARMA_Find*` versions shipped with the armadillo source code didn't appear to me to actually do much since OpenBLAS support was added to cmake in version 3.6.0. Their developers should probably up the minimum cmake requirement and drop these custom modules too). One of the things the stock FindBlas module doesn't do is set "${OpenBLAS_FOUND}"  which should be unnecessary given BLAS_FOUND and BLA_VENDOR.

If using the stock FindPKG modules is not possible, then the required external custom cmake modules should be shipped with this source code so builds of this don't need to fetch both this and armadillo sources in the case that armadillo is previously installed.

**Comment by ramkikannan** [comment]
Matt,

Thanks for the excellent suggestion. I have made the following changes.

1. Changed the cmake_minimum_required to 3.6 for all the cmake build.
2. Removed line 48-50 from planc/common/cmake_aux/NMFLIB_FindARMA.cmake
3. Included find_package(BLAS REQUIRED), find_package(LAPACK REQUIRED)
4. Made appropriate changes in planc/common/cmake_aux/NMFLIB_FindARMA.cmake

I loaded the gcc, cuda, cmake/3.6.1 and openblas modules. I ran the build.sh and everything seems to work fine in the latest commit. Let me know if I have to take care of anything more.

I also changed the release 0.8 to include this commit. I will close this issue if you see everything working.

Ramki
